sensor/upgrader/bundle/fetcher_test.go

```diff
@@ -0,0 +1,68 @@
+package bundle
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stackrox/rox/pkg/fixtures/fixtureconsts"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/sensor/upgrader/config"
+	"github.com/stackrox/rox/sensor/upgrader/upgradectx"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	expectedRequestBody = `{
+	"id": "` + fixtureconsts.Cluster1 + `"
+}`
+)
+
+func TestFetcherRequestEncoding(t *testing.T) {
+	handler := getFakeZipHandler(t, expectedRequestBody)
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	fetcherCtx, err := upgradectx.CreateForTest(
+		context.Background(),
+		t,
+		&config.UpgraderConfig{
+			ClusterID:       fixtureconsts.Cluster1,
+			CentralEndpoint: strings.TrimPrefix(server.URL, "http://"),
+		},
+	)
+	assert.NoError(t, err)
+
+	clusterFetcher := &fetcher{ctx: fetcherCtx}
+	bundle, fetchErr := clusterFetcher.FetchBundle()
+	assert.Nil(t, bundle)
+	assert.Error(t, fetchErr)
+	expectedErrorText := "making HTTP request to central for cluster bundle download: " +
+		"received response code 404 Not Found, but expected 2xx; error message: " +
+		"No data for requested cluster ID"
+	assert.Equal(t, expectedErrorText, fetchErr.Error())
```

Comment:
Wouldn't an assertion with [ErrorIs](https://pkg.go.dev/github.com/stretchr/testify/assert#ErrorIs) be a bit more robust?